components/st-microelectronics/generate-ioc.stanza

```diff
@@ -0,0 +1,351 @@
+#use-added-syntax(jitx)
+defpackage ocdb/st-microelectronics/generate-ioc:
+  import core
+  import collections
+  import jitx
+  import jitx/commands
+
+  import ocdb/checks
+  import ocdb/generic-components
+
+  import jitx/powergen/gen
+
+  import ocdb/design-vars
+  import ocdb/property-structs
+  import ocdb/bundles
+
+public pcb-enum ocdb/st-microelectronics/generate-ioc/GPIOModes :
```

Comment:
Since this is an enum and the elements are global, you may want to prefix with STM e.g. `STMGPIOOutput`

Or if you feel these should be used for everything - move the enum to somewhere common like property-structs.

Reply:
For now I think that I will prefix the enums as I don't know if these names will make sense in the global sense (yet).